python/dgl/distributed/rpc.py

```diff
@@ -766,7 +766,9 @@ def send_requests_to_machine(target_and_requests):
         service_id = request.service_id
         msg_seq = incr_msg_seq()
         client_id = get_rank()
-        server_id = target
+
+        server_id = random.randint(target*get_num_server_per_machine(),
```

Comment:
Randomly select a backup server to sent the request?

Reply:
yes. `send_request_to_machine` does the same thing.